olaf/_internals/services/logs.py

```diff
@@ -47,12 +55,6 @@ def on_loop(self):
         self.sleep(0.1)
 
     def on_read_since_boot(self) -> str:
-        """SDO callback to get a copy of logs since boot."""
-
-        if not os.path.isfile(TMP_LOGS_FILE):
-            return "no logs"
-
-        with open(TMP_LOGS_FILE, "r") as f:
-            ret = "".join(reversed(f.readlines()[-500:]))
+        """SDO callback to get a copy of the OLAF app logs since boot."""
 
-        return ret
+        return "\n".join(reversed(_logs))
```

Comment:
`_logs` is accessed from multiple threads and therefore needs an associated lock to access it to be thread safe. If this callback becomes part of `LogServiceHandler` it can reuse the Handler's existing lock which is held when `.emit()` is called.